sycl/test-e2e/syclcompat/launch/launch_policy_neg.cpp

```diff
@@ -0,0 +1,169 @@
+
+void test_variadic_config_ctor() {
+  std::cout << __PRETTY_FUNCTION__ << std::endl;
+
+  // Missing range
+  {
+    compat_exp::launch_policy missing_range_config(
+        compat_exp::kernel_properties{sycl_exp::sub_group_size<32>});
+    //CHECK-DAG: error: static assertion failed due to requirement 'syclcompat::detail::is_range_or_nd_range_v
+  }
+
+  // Duplicate nd_range
+  {
+    sycl::nd_range<3> launch_range{{1,1,32},{1,1,32}};
+    compat_exp::launch_policy duplicate_nd_range_config(launch_range, launch_range);
+    //CHECK-DAG: error: static assertion failed due to requirement 'std::conjunction_v<std::disjunction<
+  }
+
+  // Duplicate range
+  {
+    sycl::range<3> launch_range{1,1,32};
+    compat_exp::launch_policy duplicate_nd_range_config(launch_range, launch_range);
+    //CHECK-DAG: error: static assertion failed due to requirement 'std::conjunction_v<std::disjunction<
+  }
+
+  // Unwrapped property
+  {
+    sycl::nd_range<3> launch_range{{1,1,32},{1,1,32}};
+    compat_exp::launch_policy unwrapped_property_config(launch_range, {sycl_exp::sub_group_size<32>});
+    //CHECK-DAG: error: no viable constructor or deduction guide for deduction of template arguments of 'compat_exp::launch_policy'
+  }
+
+  // Foreign object in ctor
+  {
+    dummy_properties foreign_object{sycl_exp::sub_group_size<32>};
+    sycl::nd_range<3> launch_range{{1,1,32},{1,1,32}};
+    compat_exp::launch_policy unwrapped_property_config(launch_range, foreign_object);
+    //CHECK-DAG: error: no viable constructor or deduction guide for deduction of template arguments of 'compat_exp::launch_policy'
+  }
+  // Local mem with sycl::range launch 1
+  {
+    sycl::range<3> launch_range{1, 1, 32};
+    compat_exp::local_mem_size lmem_size(0);
+    compat_exp::launch_policy duplicate_local_mem_config(launch_range,
+                                                         lmem_size);
+    //CHECK-DAG: error: static assertion failed due to requirement 'syclcompat::detail::is_nd_range_v<sycl::range<3>> || !true': sycl::range kernel launches are incompatible with local
+  }
+  // Local mem with sycl::range launch 2
+  {
+    syclcompat::dim3 launch_range{32, 1, 1};
+    compat_exp::local_mem_size lmem_size(0);
+    compat_exp::launch_policy duplicate_local_mem_config(launch_range, compat_exp::kernel_properties{sycl_exp::sub_group_size<32>},
+                                                         lmem_size);
+    //CHECK-DAG: error: static assertion failed due to requirement 'syclcompat::detail::is_nd_range_v<sycl::range<3>> || !true': sycl::range kernel launches are incompatible with local
+  }
+  // Duplicate local_mem spec
+  {
+    sycl::nd_range<3> launch_range{{1, 1, 32}, {1, 1, 32}};
+    compat_exp::local_mem_size lmem_size(0);
+    compat_exp::launch_policy duplicate_local_mem_config(launch_range, lmem_size, lmem_size);
+    //CHECK-DAG: error: static assertion failed due to requirement{{.*exactly once}}
```

Comment:
This test is not failing as expected in my workstation.

Reply:
Should be resolved - I've guarded each test with a `#ifdef` so that each compilation only sees one. This makes it easier to debug these issues. I've resolved the non-failure too.